packages/swimlane/data_stream/audit_logs/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,199 @@
+      if: ctx.user?.name != null
+  - rename:
+      field: swimlane.audit_log.accountId
+      target_field: cloud.origin.account.id
+      ignore_missing: true
+  - rename:
+      field: swimlane.audit_log.tenantId
+      target_field: cloud.origin.project.id
+      ignore_missing: true
+  - rename:
+      field: swimlane.audit_log.logLevel
+      target_field: log.level
+      ignore_missing: true
+  - rename:
+      field: swimlane.audit_log.path
+      target_field: url.path
+      ignore_missing: true
+  - script:
+      lang: painless
+      description: Normalize SourceIp to a single value
+      source: >-
+        def src = ctx.swimlane?.audit_log?.sourceIp;
+        if (src != null) {
+            if (src instanceof List && src.size() > 0) {
+                ctx.swimlane.audit_log.sourceIp = src[0];
+            } else if (src instanceof String) {
+                ctx.swimlane.audit_log.sourceIp = src;
+            }
+        }
+  - grok:
+      field: swimlane.audit_log.sourceIp
+      patterns:
+        - '::ffff:%{IPV4:source.ip}'
+        - '%{IPV4:source.ip}'
+        - '%{IPV6:source.ip}' 
+      ignore_failure: true
+  - geoip:
+      field: source.ip
+      target_field: source.geo
+      ignore_missing: true
+  - rename:
+      field: swimlane.audit_log.tenant
+      target_field: cloud.origin.tenant.id
+      ignore_missing: true
+  - rename:
+      field: swimlane.audit_log.category
+      target_field: log.category
+      ignore_missing: true
+  - rename:
+      field: swimlane.audit_log.logFeatureCategory
+      target_field: log.feature_category
+      ignore_missing: true
+  - rename:
+      field: swimlane.audit_log.logSource
+      target_field: log.source
+      ignore_missing: true
+  - rename:
+      field: swimlane.audit_log.logType
+      target_field: log.type
+      ignore_missing: true
+  - rename:
+      field: swimlane.audit_log.description
+      target_field: message
+      ignore_missing: true
+  - set:
+      field: message
+      value: '{{{message}}}'
+      if: ctx.message != null && ctx.message != ''
+  - rename:
+      field: swimlane.audit_log.userId
+      target_field: user.id
+      ignore_missing: true
+  - script:
+      lang: painless
+      tag: Add ECS categorization
+      params:
+        Create:
+          type:
+            - creation
+        Read:
+          type:
+            - access
+        Update:
+          type:
+            - change
+        Delete:
+          type:
+            - deletion
+        Login:
+          type:
+            - start
+        Logout:
+          type:
+            - end
+      source: >-
+        ctx.event.kind = 'event';
+        ctx.event.type = ['info'];
+        if (ctx?.swimlane?.audit_log?.ActionType == null) {
+            return;
+        }
+        if (params.get(ctx?.swimlane?.audit_log?.ActionType) == null) {
+            return;
+        }
+        def hm = new HashMap(params.get(ctx?.swimlane?.audit_log?.ActionType));
```

```suggestion
        if (ctx.swimlane?.audit_log?.ActionType == null) {
            return;
        }
        if (params.get(ctx.swimlane?.audit_log?.ActionType) == null) {
            return;
        }
        def hm = new HashMap(params.get(ctx.swimlane?.audit_log?.ActionType));
```
`ctx` is always non-null.

---

Resolved in https://github.com/elastic/integrations/pull/13972/commits/815c09309237d43e51cdd39fc29bdb42561fccf9